Sources/Sentry/SentryTracer.m

```diff
@@ -217,6 +238,32 @@ - (void)cancelIdleTimeout
     }
 }
 
+- (void)startDeadlineTimer
+{
+    self.deadlineTimer = [self.timerWrapper
+        scheduledTimerWithTimeInterval:SENTRY_AUTO_TRANSACTION_DEADLINE
+                               repeats:NO
+                                 block:^(NSTimer *_Nonnull timer) { [self deadlineTimerFired]; }];
```

Comment:
Please use a weak reference to self here. Also add a test for that, please. It can be similar to https://github.com/getsentry/sentry-cocoa/blob/1a18683901844a2970ccfb633e4ebae565361817/Tests/SentryTests/Helper/SentryFileManagerTests.swift#L175-L189

Reply:
I've used a weak reference, but I don't know how you would want to add a test for that, the code you linked to doesn't really help as something similar doesn't work for the SentryTracer test case.